File: RecycleViewTest/app/src/main/java/com/xiakaifa/app/recycleviewtest/MainActivity.java
package com.xiakaifa.app.recycleviewtest;

import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

public class MainActivity extends AppCompatActivity {
    private MyRecycleView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recycleView);
        recyclerView.setAdapter(new MyAdapter());
        recyclerView.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
        recyclerView.start();
    }

    class MyAdapter extends RecyclerView.Adapter<ViewHolder> {

        @NonNull
        @Override
        public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
            View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item, viewGroup, false);
            return new ViewHolder(view);
        }

        @Override
        public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {

        }

        @Override
        public int getItemCount() {
            return Integer.MAX_VALUE;
        }
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
        }
    }
}

File: FillingTest/app/src/main/java/com/example/fillingtest/MainActivity.kt
package com.example.fillingtest

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
	
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_main)
		
		val layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
		recycleView.adapter = MyAdapter()
		recycleView.layoutManager = layoutManager
		
	}
	
	class MyAdapter : RecyclerView.Adapter<ViewHolder>() {
		override fun onCreateViewHolder(viewGroup: ViewGroup, p1: Int): ViewHolder {
			return ViewHolder(
				LayoutInflater.from(viewGroup.context).inflate(
					R.layout.item_layout,
					viewGroup,
					false
				)
			)
		}
		
		override fun getItemCount(): Int = 1000
		
		override fun onBindViewHolder(p0: ViewHolder, p1: Int) {
		}
		
	}
	
	class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
	
	}
}
